lib/src/doc/field.rs

```diff
@@ -31,25 +31,6 @@ impl<'a> Document<'a> {
 				let old = self.initial.pick(&k);
 				// Get the input value
 				let inp = inp.pick(&k);
-				// Check for a TYPE clause
-				if let Some(kind) = &fd.kind {
-					if !val.is_none() {
```

Comment:
Hi @hchockarprasad, I discussed the change with @finnbear but completely removing the type check seems a bit excessive. As suggested to me, I tried to change
```rust
if !val.is_none() {
```
to 
```rust
if !val.is_none_or_null() {
```
which seems to also solve the problem!

Reply:
Right! Long term, we should consider that:
- the deleted lines seem to ensure that the VALUE clause always gets a value of the "right" type for that field (or NONE)
  - one option (esp. in non-strict mode) would be to treat a conversion error - including from NULL - as a NONE
  - another option would be to only check the type if the VALUE clause cares about it (would require introspecting the value)
  - we could reconsider whether we want to guarantee that the VALUE clause is passed a previous value at all
- it should never be necessary to set a field to NULL to trigger a VALUE clause. that means the linked issue is programmer-error (should have omitted the field) or a bug (if the field is required)